packages/http-client-java/generator/http-client-generator/src/main/java/com/microsoft/typespec/http/client/generator/TypeSpecPlugin.java

```diff
@@ -298,7 +298,12 @@ public TypeSpecPlugin(EmitterOptions options, boolean sdkIntegration) {
             SETTINGS_MAP.put("use-default-http-status-code-to-exception-type-mapping", false);
             SETTINGS_MAP.put("generate-samples", false);
             SETTINGS_MAP.put("generate-tests", false);
+
+            if (options.getLicenseHeader() != null) {
+                SETTINGS_MAP.put("license-header", options.getLicenseHeader());
+            }
```

Comment:
We still set
```
// Code generated by Microsoft (R) TypeSpec Code Generator.
```
to file header, when there is no `license` configured, as line
https://github.com/microsoft/typespec/blob/76635ec7328a97b5b5717fa7146f3b9ad04823a2/packages/http-client-java/generator/http-client-generator/src/main/java/com/microsoft/typespec/http/client/generator/TypeSpecPlugin.java#L294

Should we skip this header?

Reply:
If license is set, will this line appear in header?

Reply:
Got it. In my opinion, we should be consistent regardless of whether the license is set or not.

I guess this should be consistent across languages too.

Reply:
Agree, that is the reason I am checking whether we'd remove the `Code generated` part altogether, or put it for all cases.

Reply:
We decided not to put `Code generated by Microsoft (R) TypeSpec Code Generator.` to file header